entc/gen/template/dialect/sql/query.tmpl

```diff
@@ -272,74 +271,58 @@ func ({{ $receiver }} *{{ $builder }}) sqlQuery(ctx context.Context) *sql.Select
 	{{- $receiver := $.Scope.Rec }}
 	if query := {{ $receiver }}.{{ $e.EagerLoadField }}; query != nil {
 		{{- if $e.M2M }}
-			fks := make([]driver.Value, 0, len(nodes))
-			ids := make(map[{{ $.ID.Type }}]*{{ $.Name }}, len(nodes))
-			for _, node := range nodes {
-				ids[node.ID] = node
-				fks = append(fks, node.ID)
+			edgeids := make([]driver.Value, len(nodes))
+			byid := make(map[{{ $.ID.Type }}]*{{ $.Name }})
+			nids := make(map[{{ $e.Type.ID.Type }}]map[*{{ $.Name }}]struct{})
+			for i, node := range nodes {
+				edgeids[i] = node.ID
+				byid[node.ID] = node
 				node.Edges.{{ $e.StructField }} = []*{{ $e.Type.Name }}{}
 			}
-			var (
-				edgeids []{{ $e.Type.ID.Type }}
-				edges = make(map[{{ $e.Type.ID.Type }}][]*{{ $.Name }})
-			)
-			_spec := &sqlgraph.EdgeQuerySpec{
-				Edge: &sqlgraph.EdgeSpec{
-					Inverse: {{ $e.IsInverse }},
-					Table: {{ $.Package }}.{{ $e.TableConstant }},
-					Columns: {{ $.Package }}.{{ $e.PKConstant }},
-				},
-				Predicate: func(s *sql.Selector) {
-					s.Where(sql.InValues({{ $.Package }}.{{ $e.PKConstant }}[{{ if $e.IsInverse }}1{{ else }}0{{ end }}], fks...))
-				},
+			query.Where(func(s *sql.Selector) {
+				joinT := sql.Table({{ $.Package }}.{{ $e.TableConstant }})
+				{{- $edgeid := print $e.Type.Package "." $e.Type.ID.Constant }}
+				{{- $fk1idx := 1 }}{{- $fk2idx := 0 }}{{ if $e.IsInverse }}{{ $fk1idx = 0 }}{{ $fk1idx = 1 }}{{ end }}
```

Comment:
`{{ $fk1idx = 0 }}{{ $fk1idx = 1 }}`

That part looks not right. One should be `fk2idx`, correct?

Reply:
Correct.